src/ansi-c/c_typecheck_expr.cpp

```diff
@@ -306,6 +306,8 @@ void c_typecheck_baset::typecheck_expr_main(exprt &expr)
     // replace declaration by symbol expression
     symbol_exprt bound=to_symbol_expr(expr.op0().op0());
     expr.op0().swap(bound);
+
+    implicit_typecast_bool(expr.op1());
```

Comment:
Could we have a regression test for this? I suppose that might be fairly easy to construct from one of the existing tests using quantifiers?

Reply:
Covered by existing tests; I've added a note about that into the commit message.